api/src/main/java/org/openmrs/module/eptsreports/reporting/library/cohorts/TxRttCohortQueries.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * The contents of this file are subject to the OpenMRS Public License
+ * Version 1.0 (the "License"); you may not use this file except in
+ * compliance with the License. You may obtain a copy of the License at
+ * http://license.openmrs.org
+ *
+ * Software distributed under the License is distributed on an "AS IS"
+ * basis, WITHOUT WARRANTY OF ANY KIND, either express or implied. See the
+ * License for the specific language governing rights and limitations
+ * under the License.
+ *
+ * Copyright (C) OpenMRS, LLC.  All Rights Reserved.
+ */
+package org.openmrs.module.eptsreports.reporting.library.cohorts;
+
+import java.util.Date;
+import org.openmrs.Location;
+import org.openmrs.module.eptsreports.metadata.HivMetadata;
+import org.openmrs.module.eptsreports.reporting.library.queries.TxRttQueries;
+import org.openmrs.module.eptsreports.reporting.utils.EptsReportUtils;
+import org.openmrs.module.reporting.cohort.definition.CohortDefinition;
+import org.openmrs.module.reporting.cohort.definition.CompositionCohortDefinition;
+import org.openmrs.module.reporting.evaluation.parameter.Parameter;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class TxRttCohortQueries {
+
+  private HivMetadata hivMetadata;
+
+  private GenericCohortQueries genericCohortQueries;
+
+  @Autowired
+  public TxRttCohortQueries(HivMetadata hivMetadata, GenericCohortQueries genericCohortQueries) {
+    this.hivMetadata = hivMetadata;
+    this.genericCohortQueries = genericCohortQueries;
+  }
+
+  /**
+   * For each occurred consultation (Ficha de Seguimento Adulto, Pediatria or Ficha Clinica) of
+   * Encounter Type Ids 6 or 9 during the reporting period, the occurred encounter date minus the
+   * previous scheduled consultation date (Concept ID 1410 value_datetime) from the previous
+   * consultation of Encounter Type Ids 6 or 9, is greater than 28 days
+   *
+   * @return CohortDefinition
+   */
+  public CohortDefinition getAllPatientsWhoDelayedMoreThan28Days() {
```

Comment:
I think this method  should be private as it is not being using  on somewhere else